pmm-tests/check_upgrade.py

```diff
@@ -10,117 +17,166 @@ def parse_args():
     parser.add_argument('-p', '--pre_post', help="Pass 'pre' from pre-upgrade stage and pass 'post' for post-upgrade "
                                                  "stage", default='post')
     parser.add_argument('-e', '--env', help="Pass 'ami' if it is AMI env, otherwise leave empty", default='not_ami')
-
-    args = parser.parse_args()
-
-    return args
+    return parser.parse_args()
 
 
 def verify_command(command):
-    with open("/tmp/output.log", "a") as output:
-        error_code = subprocess.call(command, shell=True, stdout=output, stderr=output)
-        if error_code != 0:
-            output = subprocess.getoutput(command.split("|")[0])
-            assert error_code == 0, f"expected to run {command} without errors. \nOutput: {output}"
-
+    """
+    Executes shell specified command and return the console output.
+    If the exit code was non-zero it raises a CalledProcessError.
+
+    :param  command:    shell command string to execute
+    :type   command:    str
+    :return:            command console output
+    :rtype:             str
+    :raises:            a CalledProcessError if the exit code was non-zero.
+    """
+    from tempfile import TemporaryFile
+    with TemporaryFile() as output:
+        return subprocess.check_output(command, shell=True, text=True, stderr=output).rstrip()
+
```

Comment:
Can we pls also add here some exception handling to catch error if any to help debug:
try:
    return subprocess.check_output.....
    except subprocess.CalledProcessError as e:
    print(f"Command failed with return code: e.returncode and output: e.output")

Reply:
no additional error hanling required, please see ghe original check_output() doc, it containt full command and return code: https://docs.python.org/3/library/subprocess.html#subprocess.check_output

Reply:
Here is example error:
```
======================================================================
ERROR: test_pmm_managed_status (__main__.PmmServerComponents)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "check_upgrade.py", line 104, in test_pmm_managed_status
    self.assertIn(RUNNING, grep_supervisor_status('pmm-managed'), NOT_RUNNING_MSG)
  File "check_upgrade.py", line 176, in grep_supervisor_status
    return verify_command(f"docker exec {pmm_server_docker_container} supervisorctl status | grep {name}")
  File "check_upgrade.py", line 36, in verify_command
    return subprocess.check_output(command, shell=True, text=True, stderr=output).rstrip()
  File "/usr/lib/python3.8/subprocess.py", line 415, in check_output
    return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
  File "/usr/lib/python3.8/subprocess.py", line 516, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command 'docker exec  supervisorctl status | grep pmm-managed' returned non-zero exit status 1.
```
I think it's enough for us... Eventually we can add more specific error message later when will find out in which case the example is not informative.